R/mechanism.R

```diff
@@ -12,7 +12,8 @@
 #' @field n Number of observations
 #' @field epsilon Differential privacy parameter
 #' @field delta Differential privacy parameter
-#' @field rng A priori estimate of the variable range
+#' @field rng A priori estimate of the variable range(s).
+#' @field rngFormat One of 'vector' (for single rng input) or 'list' (for multiple rng inputs)
```

Comment:
Just curious, what is the purpose of this new parameter?

Reply:
It flags if the range is a single range or a sequence of ranges. For backwards compatibility (which, idk, maybe a bad idea), I allow single ranges to be input as a vector, e.g. `c(0,5)`. However, you have to enter multiple ranges as a list (or matrix, though those get automatically coerced to lists) because it's the only way to allow people to enter `NULL` for a single range within the series of ranges, which is how the rest of the library treats empty things (as opposed to using `NA`s rather than `NULL`s). And in the first case, when you pass rng, rng[1]=the minimal bound of the range, and in the second, rng[1] is both the minima and the maxima of the range, so you have to treat them differently.

tldr; if we want to support single ranges entered as vectors as currently done, and support matrices/lists of ranges for more complex statistics and also support some of those ranges potentially having missing values or being completely empty/null, the easiest way to do this is to additionally pass in a parameter that describes how the range is formatted. 

side note: given the way R treats `NULL`, we maybe should actually be using `NA` in most of the places we use `NULL`, but as a computer scientist that feels uncomfortable to me, since most of the time we really do mean `NULL`, it's just that R gets fussy about that. 

It's not the prettiest but it works. There are better ways to do it but